### PR TITLE
Log in button only enabled when credentials are typed only during basic auth

### PR DESCRIPTION
**Pull request by sd1998**
Fix to issue #2302.
_____
### BUGS & IMPROVEMENTS:

- [x] (1) Log in button does not reappear after rotating the screen [FIXED]
- [X] (2) After tapping URL, Log In button disappears https://github.com/owncloud/android/pull/2309#issuecomment-444187455  [FIXED]

### CODE REVIEW

**Comment by sd1998** [comment]
@jesmrec I have fixed the issue.

**Comment by SubhrajyotiSen** [comment]
@sd1998 Maybe you can add a `trim()` after `length()` because otherwise just spaces would also be counted as valid input

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 1]
```diff
@@ -556,6 +558,28 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             }
         };
 
+        mUsernamePasswordInputWatcher = new TextWatcher(){
```
Comment (davigonz) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Include a space between `)` and `{`. Same in other parts of the class.

File: src/com/owncloud/android/authentication/AuthenticatorActivity.java [hunk 2]
```diff
@@ -556,6 +558,28 @@ public void onTextChanged(CharSequence s, int start, int before, int count) {
             }
         };
 
+        mUsernamePasswordInputWatcher = new TextWatcher(){
+
+            @Override
+            public void afterTextChanged(Editable s){
+                if(BASIC_TOKEN_TYPE.equals(mAuthTokenType)){
```
Comment (davigonz) [inline src/com/owncloud/android/authentication/AuthenticatorActivity.java]: Space between `if` and `(`

**Comment by sd1998** [comment]
@davigonz I have made the changes. Here also there seems to some changes in the owncloud-android-library which I have not made and am unable to understand why it is happening.

**Comment by sd1998** [comment]
@davigonz I have fixed it just had to rebase owncloud-android-library

**Comment by davigonz** [comment]
### (1) 

Just cloned your code and detected a tiny bug when rotating the screen

**Steps to reproduce**
1. Type a server with basic authentication.
2. Type the username.
3. Type the password, the login button appears as expected.
4. Set the screen in landscape mode by rotating the device. 

**Current behaviour**: login button doesn't appear.
**Expected behaviour**: login button should appear.

Nexus 6P v7 server 10.0.10

**Comment by sd1998** [comment]
@davigonz I have fixed the bug.

**Comment by davigonz** [comment]
Thanks @sd1998 for applying the changes, the code looks good to me and the bug seems to be fixed. Anyway, @jesmrec you can test it again.

**Comment by jesmrec** [comment]
@sd1998 branch in the app  is conflicting with android-library module. Could you fix the conflicts to pass a tiny QA stage before merging?

**Comment by sd1998** [comment]
@jesmrec I have done the rebase.

**Comment by jesmrec** [comment]
Look at these steps of a corner case:

1. Type a correct URL
2. Type incorrect credentials -> Log In button is displayed
3. Change the URL for an incorrect one -> login button is gone
4. Set again the correct URL 

Current: URL and credentials are not empty and Log In button is gone
Expected: URL and credentials are not empty and Log In button is displayed

**Comment by sd1998** [comment]
@jesmrec @davigonz I have fixed the bug.

**Comment by jesmrec** [comment]
Checks done:

- [X] Username, no password
- [X] No Username, no password
- [X] No Username, password
- [X] Wrong URL
- [X] Add new account mode
- [X] Edit account mode
- [X] URL hidden
- [X] OAuth2 URL 

Approved on my side.

I will send a PR with changes in Espresso test to fit this new set up.

**Comment by davigonz** [comment]
@hannesa2 merged, to be included in 2.10 or most likely 3.0

**Pull request merged**